builder/dockerfile/instructions/parse.go

```diff
@@ -271,16 +272,27 @@ func parseFrom(req parseRequest) (*Stage, error) {
 		return nil, err
 	}
 
+	flPlatform := req.flags.AddString("platform", "")
 	if err := req.flags.Parse(); err != nil {
 		return nil, err
 	}
+	specPlatform := system.ParsePlatform(flPlatform.Value)
+	if err := system.ValidatePlatform(specPlatform); err != nil {
+		return nil, fmt.Errorf("invalid platform %q on FROM", flPlatform.Value)
+	}
+	if specPlatform.OS != "" && !system.IsOSSupported(specPlatform.OS) {
+		return nil, fmt.Errorf("unsupported platform %q on FROM", flPlatform.Value)
+	}
+	if err != nil {
+		return nil, err
+	}
 	code := strings.TrimSpace(req.original)
-
 	return &Stage{
-		BaseName:   req.args[0],
-		Name:       stageName,
-		SourceCode: code,
-		Commands:   []Command{},
+		BaseName:        req.args[0],
+		Name:            stageName,
+		SourceCode:      code,
+		Commands:        []Command{},
+		OperatingSystem: specPlatform.OS,
```

Comment:
Since we're adding this here, why not store the `specPlatform` instead of just `OS`?

Reply:
If you see https://github.com/moby/moby/issues/34617, and the google doc linked to in the description, that's the write up of the F2F in August last year. Specifically the piece about multi-arch. The daemon is not currently multi-arch throughout, that's a follow-on that someone could do at a later time, but is out of scope for LCOW. Currently for LCOW, I have extended it for multi-OS, and have hence tried to keep it very clear in the code base when there's a "platform" (implying OS+Arch+...) as opposed to an OS. Hence why just "OperatingSystem".

Reply:
My request is not to make the daemon multi-arch.

If someone specifies `platform=linux/arm` on an amd64 system I believe this is going to silently succeed. I don't think that's correct.

We should be use platform and the `spec.Platform` type everywhere. Passing strings around is error prone and more likely to hide bugs.

Reply:
I think you may also need the entire platform spec to do the proposed default of checking against the manifest list.

Reply:
Actually no, not so. in `parseFrom` it calls system.ValidatePlatform which will error out.


```
	// Based on https://github.com/moby/moby/pull/34642#issuecomment-330375350, do
	// not support anything except operating system.
	if platform.Architecture != "" {
		return fmt.Errorf("invalid platform architecture %q", platform.Architecture)
	}
```

Hence you get the following pulling arm on amd64:
```
PS E:\docker\build\arm> docker build .
Sending build context to Docker daemon  2.048kB
Error response from daemon: Dockerfile parse error line 1: invalid platform "linux/arm" on FROM
PS E:\docker\build\arm> type .\Dockerfile
FROM --platform=linux/arm busybox
RUN echo above should fail
PS E:\docker\build\arm>
```

Reply:
Hence why deliberately using operating system as a string here until someone wants to make the entire daemon multi-arch as well.

Reply:
Why discard useless information that we are going to need?

Reply:
It isn't needed until the daemon is fully multi-arch aware, which as I say, is out of scope for LCOW. See distribution\pull_v2* function `filterManifests` for example. It only matches manifests that match `runtime.GOARCH`.